recipes/emsdk/all/conanfile.py

```diff
@@ -0,0 +1,135 @@
+import os
+from pathlib import Path
+
+from conan import ConanFile
+from conan.tools.build import cross_building
+from conan.tools.env import Environment, VirtualBuildEnv
+from conan.tools.files import chdir, copy, get
+from conan.tools.layout import basic_layout
+
+required_conan_version = ">=2.1"
+
+
+class EmSDKConan(ConanFile):
+    name = "emsdk"
+    description = "Emscripten SDK. Emscripten is an Open Source LLVM to JavaScript compiler"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/kripken/emscripten"
+    topics = ("emsdk", "emscripten", "sdk", "emcc", "em++", "nodejs")
+    license = "MIT"
+    package_type = "application"
+    settings = "os", "arch"
+    upload_policy = "skip"
+    build_policy = "missing"
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    @property
+    def _relative_paths(self):
+        return ["bin", os.path.join("bin", "upstream", "emscripten")]
+
+    @property
+    def _paths(self):
+        return [os.path.join(self.package_folder, path) for path in self._relative_paths]
+
+    @property
+    def _emsdk(self):
+        return os.path.join(self.package_folder, "bin")
+
+    @property
+    def _emscripten(self):
+        return os.path.join(self.package_folder, "bin", "upstream", "emscripten")
+
+    @property
+    def _em_config(self):
+        return os.path.join(self.package_folder, "bin", ".emscripten")
+
+    @property
+    def _em_cache(self):
+        return os.path.join(self.package_folder, "bin", ".emscripten_cache")
+
+    @property
+    def _node_path(self):
+        subfolders = [path for path in (Path(self.package_folder) / "bin" / "node").iterdir() if path.is_dir()]
+        if len(subfolders) != 1:
+            return None
+        return os.path.join("bin", "node", subfolders[0].name, "bin")
+
+    def generate(self):
+        env = Environment()
+        env.prepend_path("PATH", self._paths)
+        env.define_path("EMSDK", self._emsdk)
+        env.define_path("EMSCRIPTEN", self._emscripten)
+        env.define_path("EM_CONFIG", self._em_config)
+        env.define_path("EM_CACHE", self._em_cache)
+        env.vars(self, scope="emsdk").save_script("emsdk_env_file")
```

Comment:
It is not very clear who is using this file in this scope.

Reply:
Greetings, we have address this in the latest commit.